testing/src/test/java/org/oppia/android/testing/lightweightcheckpointing/ExplorationCheckpointTestHelperTest.kt

```diff
@@ -102,43 +169,90 @@ class ExplorationCheckpointTestHelperTest {
     assertThat(explorationCheckpointCaptor.value.isSuccess()).isTrue()
     assertThat(explorationCheckpointCaptor.value.getOrThrow().explorationTitle)
-      .isEqualTo(FAKE_EXPLORATION_TITLE_1)
+      .isEqualTo(FRACTIONS_EXPLORATION_1_TITLE)
+    assertThat(explorationCheckpointCaptor.value.getOrThrow().pendingStateName)
+      .isEqualTo(FRACTIONS_STORY_0_EXPLORATION_1_SECOND_STATE_NAME)
+  }
+
+  @Test
+  fun testSaveCheckpointForRatiosStory0Exploration0_checkCheckpointIsSaved() {
+    explorationCheckpointTestHelper.saveCheckpointForRatiosStory0Exploration0(
+      profileId = profileId,
+      version = RATIOS_STORY_0_EXPLORATION_0_CORRECT_VERSION
+    )
 
-    reset(mockExplorationCheckpointObserver)
-    retrieveFakeCheckpointLiveData =
+    val retrieveFakeCheckpointLiveData =
       explorationCheckpointController.retrieveExplorationCheckpoint(
         profileId,
-        FAKE_EXPLORATION_ID_2
+        RATIOS_EXPLORATION_ID_0
       ).toLiveData()
     retrieveFakeCheckpointLiveData.observeForever(mockExplorationCheckpointObserver)
     testCoroutineDispatchers.runCurrent()
 
-    // Verify the second exploration checkpoint was saved with correct exploration title.
+    // Verify exploration checkpoint was saved with correct exploration title.
     verify(mockExplorationCheckpointObserver, atLeastOnce())
       .onChanged(explorationCheckpointCaptor.capture())
     assertThat(explorationCheckpointCaptor.value.isSuccess()).isTrue()
     assertThat(explorationCheckpointCaptor.value.getOrThrow().explorationTitle)
-      .isEqualTo(FAKE_EXPLORATION_TITLE_2)
+      .isEqualTo(RATIOS_EXPLORATION_0_TITLE)
+  }
+
+  @Test
+  fun testUpdateCheckpointForRatiosStory0Exploration0_checkCheckpointIsSaved() {
+    explorationCheckpointTestHelper.saveCheckpointForRatiosStory0Exploration0(
+      profileId = profileId,
+      version = RATIOS_STORY_0_EXPLORATION_0_CORRECT_VERSION
+    )
+
+    explorationCheckpointTestHelper.updateCheckpointForRatiosStory0Exploration0(
+      profileId = profileId,
+      version = RATIOS_STORY_0_EXPLORATION_0_CORRECT_VERSION
+    )
+
+    val retrieveFakeCheckpointLiveData =
+      explorationCheckpointController.retrieveExplorationCheckpoint(
+        profileId,
+        RATIOS_EXPLORATION_ID_0
+      ).toLiveData()
+    retrieveFakeCheckpointLiveData.observeForever(mockExplorationCheckpointObserver)
+    testCoroutineDispatchers.runCurrent()
+
+    // Verify exploration checkpoint was saved with correct exploration title.
+    verify(mockExplorationCheckpointObserver, atLeastOnce())
+      .onChanged(explorationCheckpointCaptor.capture())
+    assertThat(explorationCheckpointCaptor.value.isSuccess()).isTrue()
+    assertThat(explorationCheckpointCaptor.value.getOrThrow().explorationTitle)
+      .isEqualTo(FRACTIONS_EXPLORATION_0_TITLE)
+    assertThat(explorationCheckpointCaptor.value.getOrThrow().pendingStateName)
+      .isEqualTo(FRACTIONS_STORY_0_EXPLORATION_0_SECOND_STATE_NAME)
```

Comment:
This & above seem odd--why does a ratios checkpoint have fractions values in it?

Reply:
updated.